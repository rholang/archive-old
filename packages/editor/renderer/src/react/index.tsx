import * as React from 'react';
// @ts-ignore: unused variable
// prettier-ignore
import { ComponentType, Consumer, Provider } from 'react';
import { Fragment, Mark, MarkType, Node, Schema } from 'prosemirror-model';
import { Serializer } from '../';
import { RendererAppearance } from '../ui/Renderer/types';
import { AnalyticsEventPayload } from '../analytics/events';

import {
  Doc,
  mergeTextNodes,
  isTextWrapper,
  TextWrapper,
  toReact,
} from './nodes';

import { toReact as markToReact } from './marks';
import {
  ProviderFactory,
  getMarksByOrder,
  isSameMark,
  EventHandlers,
  ExtensionHandlers,
  calcTableColumnWidths,
} from '@atlaskit/editor-common';
import { getText } from '../utils';

export interface RendererContext {
  objectAri?: string;
  containerAri?: string;
  adDoc?: any;
  schema?: Schema;
}

export interface ConstructorParams {
  providers?: ProviderFactory;
  eventHandlers?: EventHandlers;
  extensionHandlers?: ExtensionHandlers;
  portal?: HTMLElement;
  objectContext?: RendererContext;
  appearance?: RendererAppearance;
  disableHeadingIDs?: boolean;
  allowDynamicTextSizing?: boolean;
  allowHeadingAnchorLinks?: boolean;
  allowColumnSorting?: boolean;
  fireAnalyticsEvent?: (event: AnalyticsEventPayload) => void;
  shouldOpenMediaViewer?: boolean;
}

type MarkWithContent = Partial<Mark<any>> & {
  content: Array<MarkWithContent | Node<any>>;
};

function mergeMarks(marksAndNodes: Array<MarkWithContent | Node>) {
  return marksAndNodes.reduce(
    (acc, markOrNode) => {
      const prev = (acc.length && acc[acc.length - 1]) || null;

      if (
        markOrNode.type instanceof MarkType &&
        prev &&
        prev.type instanceof MarkType &&
        Array.isArray(prev.content) &&
        isSameMark(prev as Mark, markOrNode as Mark)
      ) {
        prev.content = mergeMarks(
          prev.content.concat((markOrNode as MarkWithContent).content),
        );
      } else {
        acc.push(markOrNode);
      }

      return acc;
    },
    [] as Array<MarkWithContent | Node>,
  );
}

export default class ReactSerializer implements Serializer<JSX.Element> {
  private providers?: ProviderFactory;
  private eventHandlers?: EventHandlers;
  private extensionHandlers?: ExtensionHandlers;
  private portal?: HTMLElement;
  private rendererContext?: RendererContext;
  private appearance?: RendererAppearance;
  private disableHeadingIDs?: boolean;
  private headingIds: string[] = [];
  private allowDynamicTextSizing?: boolean;
  private allowHeadingAnchorLinks?: boolean;
  private allowColumnSorting?: boolean;
  private fireAnalyticsEvent?: (event: AnalyticsEventPayload) => void;
  private shouldOpenMediaViewer?: boolean;

  constructor({
    providers,
    eventHandlers,
    extensionHandlers,
    portal,
    objectContext,
    appearance,
    disableHeadingIDs,
    allowDynamicTextSizing,
    allowHeadingAnchorLinks,
    allowColumnSorting,
    fireAnalyticsEvent,
    shouldOpenMediaViewer,
  }: ConstructorParams) {
    this.providers = providers;
    this.eventHandlers = eventHandlers;
    this.extensionHandlers = extensionHandlers;
    this.portal = portal;
    this.rendererContext = objectContext;
    this.appearance = appearance;
    this.disableHeadingIDs = disableHeadingIDs;
    this.allowDynamicTextSizing = allowDynamicTextSizing;
    this.allowHeadingAnchorLinks = allowHeadingAnchorLinks;
    this.allowColumnSorting = allowColumnSorting;
    this.fireAnalyticsEvent = fireAnalyticsEvent;
    this.shouldOpenMediaViewer = shouldOpenMediaViewer;
  }

  private resetState() {
    this.headingIds = [];
  }

  serializeFragment(
    fragment: Fragment,
    props: any = {},
    target: any = Doc,
    key: string = 'root-0',
    parentInfo?: { parentIsIncompleteTask: boolean; path: Array<Node> },
  ): JSX.Element | null {
    // This makes sure that we reset internal state on re-render.
    if (key === 'root-0') {
      this.resetState();
    }

    const content = ReactSerializer.getChildNodes(fragment).map(
      (node, index) => {
        if (isTextWrapper(node)) {
          return this.serializeTextWrapper(node.content);
        }

        let props;

        if (node.type.name === 'table') {
          props = this.getTableProps(node);
        } else if (node.type.name === 'date') {
          props = this.getDateProps(node, parentInfo);
        } else if (node.type.name === 'heading') {
          props = this.getHeadingProps(node, parentInfo && parentInfo.path);
        } else if (['tableHeader', 'tableRow'].indexOf(node.type.name) > -1) {
          props = this.getTableChildrenProps(node);
        } else if (node.type.name === 'media') {
          props = this.getMediaProps(node);
        } else {
          props = this.getProps(node);
        }

        let currentPath = (parentInfo && parentInfo.path) || [];

        const parentIsIncompleteTask =
          node.type.name === 'taskItem' && node.attrs.state !== 'DONE';

        let pInfo = {
          parentIsIncompleteTask,
          path: [...currentPath, node],
        };

        const serializedContent = this.serializeFragment(
          node.content,
          props,
          toReact(node),
          `${node.type.name}-${index}`,
          pInfo,
        );

        if (node.marks && node.marks.length) {
          return ([] as Array<Mark>)
            .concat(node.marks)
            .reverse()
            .reduce((acc, mark) => {
              return this.renderMark(
                markToReact(mark),
                this.getMarkProps(mark),
                `${mark.type.name}-${index}`,
                acc,
              );
            }, serializedContent);
        }

        return serializedContent;
      },
    );

    return this.renderNode(target, props, key, content);
  }

  private serializeTextWrapper(content: Node[]) {
    return ReactSerializer.buildMarkStructure(content).map((mark, index) =>
      this.serializeMark(mark, index),
    );
  }

  private serializeMark(mark: Mark, index: number = 0) {
    if (mark.type.name === 'text') {
      return (mark as any).text;
    }

    const content = ((mark as any).content || []).map(
      (child: Mark, index: number) => this.serializeMark(child, index),
    );
    return this.renderMark(
      markToReact(mark),
      this.getMarkProps(mark),
      `${mark.type.name}-${index}`,
      content,
    );
  }

  private renderNode(
    NodeComponent: ComponentType<any>,
    props: any,
    key: string,
    content: string | JSX.Element | any[] | null | undefined,
  ): JSX.Element {
    return (
      <NodeComponent key={key} {...props}>
        {content}
      </NodeComponent>
    );
  }

  private renderMark(
    MarkComponent: ComponentType<any>,
    props: any,
    key: string,
    content: any,
  ) {
    return (
      <MarkComponent key={key} {...props}>
        {content}
      </MarkComponent>
    );
  }

  private getTableChildrenProps(node: Node) {
    return {
      ...this.getProps(node),
      allowColumnSorting: this.allowColumnSorting,
    };
  }

  private getTableProps(node: Node) {
    return {
      ...this.getProps(node),
      allowColumnSorting: this.allowColumnSorting,
      columnWidths: calcTableColumnWidths(node),
      tableNode: node,
    };
  }

  private getDateProps(
    node: Node,
    parentInfo: { parentIsIncompleteTask: boolean } | undefined,
  ) {
    return {
      timestamp: node.attrs && node.attrs.timestamp,
      parentIsIncompleteTask: parentInfo && parentInfo.parentIsIncompleteTask,
    };
  }
  private getMediaProps(node: Node) {
    return {
      ...this.getProps(node),
      shouldOpenMediaViewer: this.shouldOpenMediaViewer,
    };
  }

  private getProps(node: Node) {
    return {
      text: node.text,
      providers: this.providers,
      eventHandlers: this.eventHandlers,
      extensionHandlers: this.extensionHandlers,
      portal: this.portal,
      rendererContext: this.rendererContext,
      serializer: this,
      content: node.content ? node.content.toJSON() : undefined,
      allowDynamicTextSizing: this.allowDynamicTextSizing,
      allowHeadingAnchorLinks: this.allowHeadingAnchorLinks,
      rendererAppearance: this.appearance,
      fireAnalyticsEvent: this.fireAnalyticsEvent,
      ...node.attrs,
    };
  }

  private headingAnchorSupported(path: Array<Node> = []): boolean {
    return (
      path.length === 0 || path[path.length - 1].type.name === 'layoutColumn'
    );
  }

  private getHeadingProps(node: Node, path: Array<Node> = []) {
    return {
      ...node.attrs,
      content: node.content ? node.content.toJSON() : undefined,
      headingId: this.getHeadingId(node),
      showAnchorLink:
        this.allowHeadingAnchorLinks &&
        !this.disableHeadingIDs &&
        this.headingAnchorSupported(path),
    };
  }

  // The return value of this function is NOT url encoded,
  // In HTML5 standard, id can contain any characters, encoding is no necessary.
  // Plus we trying to avoid double encoding, therefore we leave the value as is.
  // Remember to use encodeURIComponent when generating url from the id value.
  private getHeadingId(node: Node) {
    if (this.disableHeadingIDs || !node.content.size) {
      return;
    }

    // We are not use node.textContent here, because we would like to handle cases where
    // headings only contain inline blocks like emoji, status and date.
    const nodeContent = (node as any).content
      .toJSON()
      .reduce((acc: string, node: any) => acc.concat(getText(node) || ''), '')
      .trim()
      .replace(/\s/g, '-');

    if (!nodeContent) {
      return;
    }

    return this.getUniqueHeadingId(nodeContent);
  }

  private getUniqueHeadingId(baseId: string, counter = 0): string {
    if (counter === 0 && this.headingIds.indexOf(baseId) === -1) {
      this.headingIds.push(baseId);
      return baseId;
    } else if (counter !== 0) {
      const headingId = `${baseId}.${counter}`;
      if (this.headingIds.indexOf(headingId) === -1) {
        this.headingIds.push(headingId);
        return headingId;
      }
    }

    return this.getUniqueHeadingId(baseId, ++counter);
  }

  private getMarkProps(mark: Mark): any {
    const { key, ...otherAttrs } = mark.attrs;
    return {
      eventHandlers: this.eventHandlers,
      markKey: key,
      ...otherAttrs,
    };
  }

  static getChildNodes(fragment: Fragment): (Node | TextWrapper)[] {
    const children: Node[] = [];
    fragment.forEach(node => {
      children.push(node);
    });
    return mergeTextNodes(children) as Node[];
  }

  static getMarks(node: Node): Mark[] {
    if (!node.marks || node.marks.length === 0) {
      return [];
    }

    return getMarksByOrder(node.marks);
  }

  static buildMarkStructure(content: Node[]) {
    return mergeMarks(
      content.map(node => {
        const nodeMarks = this.getMarks(node);
        if (nodeMarks.length === 0) {
          return node;
        }

        return nodeMarks.reverse().reduce(
          (acc, mark) => {
            const { eq } = mark;

            return {
              ...mark,
              eq,
              content: [acc],
            };
          },
          node as any,
        );
      }),
    ) as Mark[];
  }

  static fromSchema(
    _schema: Schema,
    {
      providers,
      eventHandlers,
      extensionHandlers,
      appearance,
      disableHeadingIDs,
      allowDynamicTextSizing,
      allowHeadingAnchorLinks,
      allowColumnSorting,
    }: ConstructorParams,
  ): ReactSerializer {
    // TODO: Do we actually need the schema here?
    return new ReactSerializer({
      providers,
      eventHandlers,
      extensionHandlers,
      appearance,
      disableHeadingIDs,
      allowDynamicTextSizing,
      allowHeadingAnchorLinks,
      allowColumnSorting,
    });
  }
}
