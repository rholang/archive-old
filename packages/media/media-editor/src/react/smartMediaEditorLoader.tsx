import * as React from 'react';
import { ModalSpinner } from '@atlaskit/media-ui';
import { WithMediaClientConfigProps } from '@atlaskit/media-client';
import { N700A } from '@atlaskit/theme/colors';
import { SmartMediaEditorProps } from './smartMediaEditor';

type SmartEditorWithMediaClientConfigProps = WithMediaClientConfigProps<
  SmartMediaEditorProps
>;
type SmartEditorWithMediaClientConfigComponent = React.ComponentType<
  SmartEditorWithMediaClientConfigProps
>;

interface AsyncSmartMediaEditorState {
  SmartMediaEditor?: SmartEditorWithMediaClientConfigComponent;
}

export default class AsyncSmartMediaEditor extends React.PureComponent<
  SmartEditorWithMediaClientConfigProps & AsyncSmartMediaEditorState,
  AsyncSmartMediaEditorState & { isErrored: boolean }
> {
  static displayName = 'AsyncSmartMediaEditor';
  static SmartMediaEditor?: SmartEditorWithMediaClientConfigComponent;

  state = {
    // Set state value to equal to current static value of this class.
    SmartMediaEditor: AsyncSmartMediaEditor.SmartMediaEditor,
    isErrored: false,
  };

  async UNSAFE_componentWillMount() {
    if (!this.state.SmartMediaEditor) {
      try {
        const [mediaClient, smartEditorModule] = await Promise.all([
          import(/* webpackChunkName:"@atlaskit-media-client" */ '@atlaskit/media-client'),
          import(/* webpackChunkName:"@atlaskit-internal_smart-media-editor" */ './smartMediaEditor'),
        ]);
        AsyncSmartMediaEditor.SmartMediaEditor = mediaClient.withMediaClient(
          smartEditorModule.default,
        );
        this.setState({
          SmartMediaEditor: AsyncSmartMediaEditor.SmartMediaEditor,
        });
      } catch (e) {
        // eslint-disable-next-line no-console
        console.error(e);
        this.setState({ isErrored: true });
      }
    }
  }

  render() {
    const { isErrored } = this.state;

    if (isErrored) {
      return null;
    }
    if (!this.state.SmartMediaEditor) {
      return <ModalSpinner blankedColor={N700A} invertSpinnerColor={true} />;
    }

    return <this.state.SmartMediaEditor {...this.props} />;
  }
}
