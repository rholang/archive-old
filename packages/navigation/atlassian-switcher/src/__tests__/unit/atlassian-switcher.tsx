import * as React from 'react';
import { shallow } from 'enzyme';
import AtlassianSwitcher from '../../components/atlassian-switcher';

const noop = () => void 0;

describe('Atlassian Switcher', () => {
  it('basic snapshot', () => {
    const switcher = shallow(
      <AtlassianSwitcher
        product="jira"
        cloudId="CLOUD_ID"
        triggerXFlow={noop}
      />,
    );
    expect(switcher).toMatchSnapshot();
  });
});
