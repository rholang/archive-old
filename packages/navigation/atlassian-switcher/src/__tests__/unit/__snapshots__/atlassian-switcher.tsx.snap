// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atlassian Switcher basic snapshot 1`] = `
<InjectIntl(SwitcherIntlProdiver)>
  <NavigationAnalyticsContext
    data={
      Object {
        "attributes": Object {
          "featureFlags": Object {
            "disableCustomLinks": false,
            "disableHeadings": false,
            "disableRecentContainers": false,
            "isDiscoverMoreForEveryoneEnabled": false,
            "isDiscoverSectionEnabled": false,
            "isEmceeLinkEnabled": false,
            "xflow": true,
          },
        },
        "componentName": "atlassianSwitcher",
        "packageName": "@atlaskit/atlassian-switcher",
        "packageVersion": "999.9.9",
        "source": "atlassianSwitcher",
      }
    }
  >
    <WithAnalyticsEvents(InjectIntl(ErrorBoundary))
      messages={
        Object {
          "administration": Object {
            "defaultMessage": "Administration",
            "description": "The text of a link redirecting the user to the site administration",
            "id": "fabric.atlassianSwitcher.administration",
          },
          "browseApps": Object {
            "defaultMessage": "Browse Marketplace apps",
            "description": "The text of a link redirecting the user to Discover Embedded Marketplace within in the product (Marketplace is a brand name. Please dont translate it)",
            "id": "fabric.atlassianSwitcher.browseApps",
          },
          "confluenceSpace": Object {
            "defaultMessage": "Confluence space",
            "description": "In a context in which several items are listed , this text describes that the specific type of a given item is a Confluence space",
            "id": "fabric.atlassianSwitcher.confluenceSpace",
          },
          "discover": Object {
            "defaultMessage": "Discover",
            "description": "The header of \\"Discover\\" section",
            "id": "fabric.atlassianSwitcher.discover",
          },
          "discoverMore": Object {
            "defaultMessage": "More Atlassian products",
            "description": "The text of a link redirecting the user to Discover More Atlassian products",
            "id": "fabric.atlassianSwitcher.discoverMore",
          },
          "errorHeading": Object {
            "defaultMessage": "Something’s gone wrong",
            "description": "Heading of the error screen which is shown when an unknown error happens in the Atlassian Switcher. Usually due to failed network requests.",
            "id": "fabric.atlassianSwitcher.errorHeading",
          },
          "errorImageAltText": Object {
            "defaultMessage": "A broken robot and a number of people busy fixing it.",
            "description": "Text displayed as alt text when an error occurs in the Atlassian Switcher",
            "id": "fabric.atlassianSwitcher.errorImageAltText",
          },
          "errorText": Object {
            "defaultMessage": "We keep track of these errors, but feel free to contact us if refreshing doesn’t fix things",
            "description": "Text that is displayed when an unknown error happens in the Atlassian Switcher.",
            "id": "fabric.atlassianSwitcher.errorText",
          },
          "errorTextLoggedOut": Object {
            "defaultMessage": "You’ve been logged out. Please log in again.",
            "description": "Text that is displayed when we detect user is logged out.",
            "id": "fabric.atlassianSwitcher.errorTextLoggedOut",
          },
          "errorTextNetwork": Object {
            "defaultMessage": "We couldn’t load this list. Please reload the page and try again.",
            "description": "Text that is displayed when we detect a network issue.",
            "id": "fabric.atlassianSwitcher.errorTextNetwork",
          },
          "jiraProject": Object {
            "defaultMessage": "Jira project",
            "description": "In a context in which several items are listed , this text describes that the specific type of a given item is a Jira project",
            "id": "fabric.atlassianSwitcher.jiraProject",
          },
          "login": Object {
            "defaultMessage": "Log in",
            "description": "Text in log in button.",
            "id": "fabric.atlassianSwitcher.login",
          },
          "manageList": Object {
            "defaultMessage": "Manage list",
            "description": "This text is for the action for a user to manage the values present in an editable list of links.",
            "id": "fabric.atlassianSwitcher.manageList",
          },
          "more": Object {
            "defaultMessage": "More",
            "description": "In a context in which users are able to view predefined custom links, this text is the title of the section displaying all existing custom links.",
            "id": "fabric.atlassianSwitcher.more",
          },
          "productDescriptionConfluence": Object {
            "defaultMessage": "Document collaboration",
            "description": "Text displayed under Confluence product recommendation.",
            "id": "fabric.atlassianSwitcher.product.description.confluence",
          },
          "productDescriptionJiraServiceDesk": Object {
            "defaultMessage": "IT service desk and customer service",
            "description": "Text displayed under Jira Service Desk product recommendation.",
            "id": "fabric.atlassianSwitcher.product.description.jsd",
          },
          "productDescriptionJiraSoftware": Object {
            "defaultMessage": "Project and issue tracking",
            "description": "Text displayed under Jira Software product recommendation.",
            "id": "fabric.atlassianSwitcher.product.description.jsw",
          },
          "productDescriptionOpsgenie": Object {
            "defaultMessage": "Modern incident management",
            "description": "Text displayed under Opsgenie product recommendation.",
            "id": "fabric.atlassianSwitcher.product.description.opsgenie",
          },
          "recent": Object {
            "defaultMessage": "Recent",
            "description": "In a context in which users are able to view recent projects or spaces they've viewed, this text is the title of the section displaying all the recent projects or spaces.",
            "id": "fabric.atlassianSwitcher.recent",
          },
          "showMoreSites": Object {
            "defaultMessage": "Show more sites",
            "description": "The text of a toggle showing more site options",
            "id": "fabric.atlassianSwitcher.show.more.sites",
          },
          "switchTo": Object {
            "defaultMessage": "Switch to",
            "description": "In a context in which users are able to switch between products, this text is the title of the category displaying the products the user is able to switch to.",
            "id": "fabric.atlassianSwitcher.switchTo",
          },
          "switchToTooltip": Object {
            "defaultMessage": "Switch to…",
            "description": "This text appears as a tooltip when a user hovers over the atlassian switcher icon before clicking on it.",
            "id": "fabric.atlassianSwitcher.switchToTooltip",
          },
          "try": Object {
            "defaultMessage": "Try",
            "description": "This text appears as a way to encourage the user to try a new Atlassian product.",
            "id": "fabric.atlassianSwitcher.try",
          },
        }
      }
    >
      <LoadableComponent
        cloudId="CLOUD_ID"
        features={
          Object {
            "disableCustomLinks": false,
            "disableHeadings": false,
            "disableRecentContainers": false,
            "isDiscoverMoreForEveryoneEnabled": false,
            "isDiscoverSectionEnabled": false,
            "isEmceeLinkEnabled": false,
            "xflow": true,
          }
        }
        messages={
          Object {
            "administration": Object {
              "defaultMessage": "Administration",
              "description": "The text of a link redirecting the user to the site administration",
              "id": "fabric.atlassianSwitcher.administration",
            },
            "browseApps": Object {
              "defaultMessage": "Browse Marketplace apps",
              "description": "The text of a link redirecting the user to Discover Embedded Marketplace within in the product (Marketplace is a brand name. Please dont translate it)",
              "id": "fabric.atlassianSwitcher.browseApps",
            },
            "confluenceSpace": Object {
              "defaultMessage": "Confluence space",
              "description": "In a context in which several items are listed , this text describes that the specific type of a given item is a Confluence space",
              "id": "fabric.atlassianSwitcher.confluenceSpace",
            },
            "discover": Object {
              "defaultMessage": "Discover",
              "description": "The header of \\"Discover\\" section",
              "id": "fabric.atlassianSwitcher.discover",
            },
            "discoverMore": Object {
              "defaultMessage": "More Atlassian products",
              "description": "The text of a link redirecting the user to Discover More Atlassian products",
              "id": "fabric.atlassianSwitcher.discoverMore",
            },
            "errorHeading": Object {
              "defaultMessage": "Something’s gone wrong",
              "description": "Heading of the error screen which is shown when an unknown error happens in the Atlassian Switcher. Usually due to failed network requests.",
              "id": "fabric.atlassianSwitcher.errorHeading",
            },
            "errorImageAltText": Object {
              "defaultMessage": "A broken robot and a number of people busy fixing it.",
              "description": "Text displayed as alt text when an error occurs in the Atlassian Switcher",
              "id": "fabric.atlassianSwitcher.errorImageAltText",
            },
            "errorText": Object {
              "defaultMessage": "We keep track of these errors, but feel free to contact us if refreshing doesn’t fix things",
              "description": "Text that is displayed when an unknown error happens in the Atlassian Switcher.",
              "id": "fabric.atlassianSwitcher.errorText",
            },
            "errorTextLoggedOut": Object {
              "defaultMessage": "You’ve been logged out. Please log in again.",
              "description": "Text that is displayed when we detect user is logged out.",
              "id": "fabric.atlassianSwitcher.errorTextLoggedOut",
            },
            "errorTextNetwork": Object {
              "defaultMessage": "We couldn’t load this list. Please reload the page and try again.",
              "description": "Text that is displayed when we detect a network issue.",
              "id": "fabric.atlassianSwitcher.errorTextNetwork",
            },
            "jiraProject": Object {
              "defaultMessage": "Jira project",
              "description": "In a context in which several items are listed , this text describes that the specific type of a given item is a Jira project",
              "id": "fabric.atlassianSwitcher.jiraProject",
            },
            "login": Object {
              "defaultMessage": "Log in",
              "description": "Text in log in button.",
              "id": "fabric.atlassianSwitcher.login",
            },
            "manageList": Object {
              "defaultMessage": "Manage list",
              "description": "This text is for the action for a user to manage the values present in an editable list of links.",
              "id": "fabric.atlassianSwitcher.manageList",
            },
            "more": Object {
              "defaultMessage": "More",
              "description": "In a context in which users are able to view predefined custom links, this text is the title of the section displaying all existing custom links.",
              "id": "fabric.atlassianSwitcher.more",
            },
            "productDescriptionConfluence": Object {
              "defaultMessage": "Document collaboration",
              "description": "Text displayed under Confluence product recommendation.",
              "id": "fabric.atlassianSwitcher.product.description.confluence",
            },
            "productDescriptionJiraServiceDesk": Object {
              "defaultMessage": "IT service desk and customer service",
              "description": "Text displayed under Jira Service Desk product recommendation.",
              "id": "fabric.atlassianSwitcher.product.description.jsd",
            },
            "productDescriptionJiraSoftware": Object {
              "defaultMessage": "Project and issue tracking",
              "description": "Text displayed under Jira Software product recommendation.",
              "id": "fabric.atlassianSwitcher.product.description.jsw",
            },
            "productDescriptionOpsgenie": Object {
              "defaultMessage": "Modern incident management",
              "description": "Text displayed under Opsgenie product recommendation.",
              "id": "fabric.atlassianSwitcher.product.description.opsgenie",
            },
            "recent": Object {
              "defaultMessage": "Recent",
              "description": "In a context in which users are able to view recent projects or spaces they've viewed, this text is the title of the section displaying all the recent projects or spaces.",
              "id": "fabric.atlassianSwitcher.recent",
            },
            "showMoreSites": Object {
              "defaultMessage": "Show more sites",
              "description": "The text of a toggle showing more site options",
              "id": "fabric.atlassianSwitcher.show.more.sites",
            },
            "switchTo": Object {
              "defaultMessage": "Switch to",
              "description": "In a context in which users are able to switch between products, this text is the title of the category displaying the products the user is able to switch to.",
              "id": "fabric.atlassianSwitcher.switchTo",
            },
            "switchToTooltip": Object {
              "defaultMessage": "Switch to…",
              "description": "This text appears as a tooltip when a user hovers over the atlassian switcher icon before clicking on it.",
              "id": "fabric.atlassianSwitcher.switchToTooltip",
            },
            "try": Object {
              "defaultMessage": "Try",
              "description": "This text appears as a way to encourage the user to try a new Atlassian product.",
              "id": "fabric.atlassianSwitcher.try",
            },
          }
        }
        product="jira"
        triggerXFlow={[Function]}
      />
    </WithAnalyticsEvents(InjectIntl(ErrorBoundary))>
  </NavigationAnalyticsContext>
</InjectIntl(SwitcherIntlProdiver)>
`;
